### PR TITLE
[ONNX] Fix expand_as

### LINKED ISSUE #95961: torch.onnx.exporter optimization for expand_as causes error in runtime
### 🐛 Describe the bug

The exported onnx graph executed wrong with error:
onnxruntime.capi.onnxruntime_pybind11_state.InvalidArgument: [ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Non-zero status code returned while running Expand node. Name:'/Expand' Status Message: invalid expand shape

This may because the optimization of expand_as in https://github.com/pytorch/pytorch/commit/f0054e1a6e04feb2a1a791d5c02da17dadf6d904 

A simple example to reproduce the error:

```
import torch
import onnxruntime

class Model(torch.nn.Module):
  def forward(self, x):
    aa = torch.tensor([[0],[1],[2]])
    return aa.expand_as(x)

x = torch.ones(3,2)
model = Model()
print(model(x))
model_path='./model.onnx.pb'
torch.onnx.export(model,(x,),model_path, input_names=["x"],dynamic_axes={"x": [0, 1]})


ort_session = onnxruntime.InferenceSession(model_path, providers=['CPUExecutionProvider'])

ort_inputs = { 'x': x.numpy() }
output_columns = []
ort_outs = ort_session.run(output_columns, ort_inputs)
```

### Versions

Collecting environment information...
PyTorch version: 1.14.0.dev20221107+cu116
Is debug build: False
CUDA used to build PyTorch: 11.6
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.24.0
Libc version: glibc-2.31

Python version: 3.8.13 (default, Mar 28 2022, 11:38:47)  [GCC 7.5.0] (64-bit runtime)
Python platform: Linux-5.0.0-1032-azure-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 11.6.124

### PR DESCRIPTION
**Pull request by guyang3532**
Fixes [#ISSUE_NUMBER](https://github.com/pytorch/pytorch/issues/95961)

### CODE REVIEW

**Comment by BowenBao** [comment]
Hello @guyang3532 , thanks for contribution! Could you please add the scenario you described as a test case? Similar to https://github.com/pytorch/pytorch/blob/7f5f0b3665191d65471b07d2dbdfc5eed515d988/test/onnx/test_pytorch_onnx_onnxruntime.py#L4116

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -941,6 +941,8 @@ def expand_as(g: jit_utils.GraphContext, self, other):
             if torch.equal(self_t.mean(d).unsqueeze(d).expand_as(self_t), self_t):
                 dims.append(d)
                 self = g.op("Constant", value_t=self_t.mean(dims).to(orig_type))
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset9.py:943]:
```suggestion
                self = g.op("Constant", value_t=self_t.mean(dims, keepdims=True).to(orig_type))
```

Looking at this again, I think this is the right fix.

Comment (guyang3532) [reply torch/onnx/symbolic_opset9.py]: Yeah, thanks a lot! Updated.

**Comment by guyang3532** [comment]
> Hello @guyang3532 , thanks for contribution! Could you please add the scenario you described as a test case? Similar to
> 
> https://github.com/pytorch/pytorch/blob/7f5f0b3665191d65471b07d2dbdfc5eed515d988/test/onnx/test_pytorch_onnx_onnxruntime.py#L4116

Added. Thanks.

**Review by BowenBao — APPROVED** [review]
LGTM, please check linter issues.
checkout https://github.com/pytorch/pytorch/wiki/lintrunner to fix the lint errors.

**Review by justinchuby — APPROVED** [review]

**Pull request closed without merging**